specs/decimal/decimal.md

```diff
@@ -45,7 +57,152 @@ data structure may be re-expressed in each language using language-specific cons
 defines the `DecimalValue` structure as the Win32 `DECIMAL` definition syntax is incompatible with
 WinRT IDL.
 
-# 3. WinRT API
+# 3. Examples
+
+The examples are a program displaying the results of the the mathematical formula:
+
+```
+x = (((a - b) + (c % d)) * e).round(5) / f;
+y = x++.round(3).clamp(f, g);
+z = --y.floor() * ++x.ceil() * -y.truncate();
+```
+
+Given the values
+
+```
+a = 1
+b = 0.5
+c = -6.66
+d = 1.23
+e = 4567.089
+f = -4
+g = 1967
+h = 1001.0
+```
+
+the output is
+
+```
+x == 12.4177225
+y == 11.418
+z == -1540
+```
+
+and the process exit code is -1540.
+
+# 4. C#
+
+NOTE: This example uses [C#'s Decimal support](https://learn.microsoft.com/dotnet/api/system.decimal)
+providing the canonical reference for examples in other languages.
+
+```c++
+Using System;
+Using System.IO;
+
+class Program
+{
+    static void Main()
+    {
+        var a = new Decimal(1);
+        var b = new Decimal(0.5);
+        var c = Decimal.Parse("-6.66");
+        var d = Decimal.Parse("1.23");
+        var e = Decimal.Parse("4567.089");
+        var f = new Decimal((long)-4);
+        var g = new Decimal((uint)1967);
+        var h = new Decimal(1001.0);
+
+        var x = Decimal.Round(((a - b) + (c % d)) * e, 5) / f;
+        var y = Math.Clamp(Decimal.Round(x++, 3), f, g);
+        var yfloor = Decimal.Floor(y);
+        var xceil = Decimal.Ceiling(x);
+        var z = --yfloor * ++xceil * -Decimal.Truncate(y);
+
+        Console.WriteLine($"x = {x}");
+        Console.WriteLine($"y = {y}");
+        Console.WriteLine($"z = {z}");
+
+        Environment.Exit((int)z);
+    }
+}
+```
+
+# 5. C++
+
+This program illustrates the reference example using Windows App SDK's C++ decimal class.
+
+```c++
+#include <windows.h>
+#include <stdio.h>
+#include <decimal.h>
+
+int main()
+{
+    const Microsoft::Windows::Foundation::decimal a{ 1 };
+    const Microsoft::Windows::Foundation::decimal b{ 0.5 };
+    const Microsoft::Windows::Foundation::decimal c{ L"-6.66" };
+    const Microsoft::Windows::Foundation::decimal d{ L"1.23" };
+    const Microsoft::Windows::Foundation::decimal e{ L"4567.089" };
+    const Microsoft::Windows::Foundation::decimal f{ -4ll };
+    const Microsoft::Windows::Foundation::decimal g{ 1967u };
+    const Microsoft::Windows::Foundation::decimal h{ 1001.0f };
+
+    auto x = (((a - b) + (c % d)) * e).round(5) / f;
+    auto y = x++.round(3).clamp(f, g);
+    auto z = --y.floor() * ++x.ceil() * -y.truncate();
+
+    WEX::Logging::Log::Comment(WEX::Common::String().Format(L"x = %ls", x.to_string().c_str()));
```

Comment:
>W [](http://example.com/codeflow?start=4&length=1)

I don't think WEX would be a well-known concept. Maybe just stick with the printfs?